src/main/index.js

```diff
@@ -68,14 +73,52 @@ if (appConfig.appType === 'development') {
 	}
 }
 
+// NOTE: Has to be set up after user data directory is updated
 const configStore = createConfigStore()
 
+let sentryUser = configStore.get('sentryUser')
+
+// NOTE: The retrieved value may be the default based on config store schema,
+// which is not immediately persisted upon initialization.
+// We want to make sure that this value is persisted upon startup, even if it might be
+// rotated shortly after.
+configStore.set('sentryUser', sentryUser)
+
+if (shouldRotateSentryUser(sentryUser)) {
+	log('Rotating Sentry user')
+	const newSentryUser = generateSentryUser()
+	configStore.set('sentryUser', newSentryUser)
+	sentryUser = newSentryUser
+}
+
+/** @type {import('../shared/app.js').SentryEnvironment} */
+let sentryEnvironment = 'development'
+
+if (appConfig.appType === 'release-candidate') {
+	sentryEnvironment = 'qa'
+} else if (appConfig.appType === 'production') {
+	sentryEnvironment = 'production'
+}
+
+// NOTE: Has to be set up after user data directory is updated
+// https://docs.sentry.io/platforms/javascript/guides/electron/#app-userdata-directory
+Sentry.init({
+	dsn: 'https://f7336c12cc39fb0367886e31036a6cd7@o4507148235702272.ingest.us.sentry.io/4509803831820288',
+	// TODO: Enable tracing based on user consent in production
+	tracesSampleRate: sentryEnvironment === 'production' ? 0 : 1.0,
+	environment: sentryEnvironment,
+	release: appConfig.appVersion,
+	debug: sentryEnvironment === 'development',
+	initialScope: { user: { id: sentryUser.id } },
```

Comment:
would be helpful to have some confirmation that this configured somewhat correctly

Reply:
Thanks, any comments on some of the options that are missing here but are done in the mobile backend's integration? Specifically looking at `enabled`, `sendDefaultPii`, `_experiments`, and the console logging integration.

https://github.com/digidem/comapeo-mobile/blob/fef201da8dcdf93e99c8a567d62ca4834b1a2b9b/src/backend/loader.js#L47-L63

Reply:
Regarding enabled it'd be good to have similar semantics to mobile for when it's on/off

I'd set `sendDefaultPii: false` unless you have reason to do otherwise.

Regarding the experiments beforeSendLog, see if the user id is being set in the sentry dashboard with your existing setup first. I had to do that because node.js sentry didn't have a concept of a "user" for logs